scripts/cnn_variant_wdl/cram2filtered.wdl

```diff
@@ -0,0 +1,511 @@
+
+
+    # Mem is in units of GB but our command and memory runtime values are in MB
+    Int machine_mem = if defined(mem_gb) then mem_gb *1000 else default_ram_mb
+    Int command_mem = machine_mem - 1000
+
+command <<<
+  ls -ltr ${cram_file} ${reference_fasta} &&
+  echo "ls (1): complete" &&
+  samtools view -h -T ${reference_fasta} ${cram_file} |
+  samtools view -b -o ${output_prefix}.bam - &&
+  echo "samtools view: complete" &&
+  ls -ltr . &&
+  echo "ls (2): complete" &&
+  samtools index -b ${output_prefix}.bam &&
+  echo "samtools index: complete" &&
+  ls -ltr . &&
+  echo "ls (3): complete" &&
+  mv ${output_prefix}.bam.bai ${output_prefix}.bai &&
+  echo "mv: complete" &&
+  ls -ltr . &&
+  echo "ls (4): complete"
+  >>>
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.1.1"
+    memory: machine_mem + " MB"
+
+    # Note that the space before SSD and HDD should be included.
+    disks: "local-disk " + sub(disk_space_gb, "\\..*", "") + " HDD"
+    preemptible: select_first([preemptible_attempts, 3])
+    cpu: select_first([cpu, 1])  
+  }
+
+  output {
+    File output_bam = "${output_prefix}.bam"
+    File output_bam_index = "${output_prefix}.bai"
+  }
+}
+
+task RunHC4 {
+    String input_bam
+    File input_bam_index
+    File reference_fasta
+    File reference_dict
+    File reference_fasta_index
+    String output_prefix
+    File interval_list
+    String extra_args
+    File? gatk_override
+
+    # Runtime parameters
+    Int? mem_gb
+    String gatk_docker
+    Int? preemptible_attempts
+    Float disk_space_gb
+    Int? cpu 
+
+    # You may have to change the following two parameter values depending on the task requirements
+    Int default_ram_mb = 8000
+
+    # Mem is in units of GB but our command and memory runtime values are in MB
+    Int machine_mem = if defined(mem_gb) then mem_gb *1000 else default_ram_mb
+    Int command_mem = machine_mem - 1000
+
+    command {
+        set -e
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk_override}
+
+        gatk --java-options "-Xmx${command_mem}m" \
+        HaplotypeCaller \
+        -R ${reference_fasta} \
+        -I ${input_bam} \
+        -O ${output_prefix}_hc4.vcf.gz \
+        -L ${interval_list} \
+        -bamout ${output_prefix}_bamout.bam \
+        ${extra_args}
+    }
+
+    output {
+        File bamout = "${output_prefix}_bamout.bam"
+        File bamout_index = "${output_prefix}_bamout.bai"
+        File raw_vcf = "${output_prefix}_hc4.vcf.gz"
+        File raw_vcf_index = "${output_prefix}_hc4.vcf.gz.tbi"
+    }
+    runtime {
+        docker: gatk_docker
+        memory: machine_mem + " MB"
+        # Note that the space before SSD and HDD should be included.
+        disks: "local-disk " + sub(disk_space_gb, "\\..*", "") + " HDD"
+        preemptible: select_first([preemptible_attempts, 3])
+        cpu: select_first([cpu, 1])
+        zones: "us-east4-a"
+        bootDiskSizeGb: "16"
+    }
+}
+
+task CNNScoreVariants {
```

Comment:
It makes sense to pull the Score and Filter out into a separate tasks WDL like the CNV workflows do: https://github.com/broadinstitute/gatk/blob/master/scripts/cnv_wdl/cnv_common_tasks.wdl  Maybe split intervals and merge too.

Reply:
done

Reply:
I'm still seeing two different files with a CNNScoreVariants task.